dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -67,6 +83,52 @@ class TypedDataset[T](
     ): GroupedByManyOps[T, K, Out0] = new GroupedByManyOps[T, K, Out0](self, groupedBy)
   }
 
+  def join[A: TypedEncoder, B](
+    right: TypedDataset[A],
+    leftCol: TypedColumn[T, B],
```

Comment:
Yeah I had the same reasoning, in the `TypedFrame` stuff I implemented [this syntax](https://github.com/adelbertc/frameless/blob/master/dataframe/src/test/scala/JoinTests.scala#L80) for equality joins, and a more concise one for `join using`.